builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -189,7 +189,28 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		statement, err = findLambdaPolicyStatementById(&policy, d.Id())
 		return resource.RetryableError(err)
 	})
+
 	if err != nil {
+		switch e := err.(type) {
+		case *resource.TimeoutError:
+			switch underlying := e.LastError.(type) {
+			case awserr.Error:
+				if underlying.Code() == "ResourceNotFoundException" {
+					log.Printf("[WARN] No Lambda Permission Policy found: %v", input)
+					d.SetId("")
+					return nil
+				}
+			case *resource.NotFoundError:
+				log.Printf("[WARN] found Lambda Permission Policy for %v but statement '%s' is absent", input, d.Id())
+				d.SetId("")
+				return nil
+			}
+		case (*resource.NotFoundError):
```

Comment:
I personally think the `switch` here is making it unnecessarily complicated... two if conditionals would IMO do the job.

Also `resource.NotFoundError` will always come from `resource.TimeoutError.LastError` since errors from `findLambdaPolicyStatementById` are always treated as `resource.RetryableError`s, so `case (*resource.NotFoundError):` will never be executed I think.